code/controllers/subsystem/bluespace_exploration.dm

```diff
@@ -0,0 +1,542 @@
+	var/list/tracked_ships
+
+	//=====Ship Processing Queue=====
+	// ! Tracks a list of all the ships requesting transit
+	// ! Key (string) - shuttle_id
+	// ! Value (/datum/data_holder/bluespace_exploration) - generation settings
+	var/list/ship_traffic_queue	//The queue for generation is disguised as 'bluespace traffic control'
+
+	//=====Z-Level Wiping=====
+	//Are we currently processing a z-level wipe
+	var/wiping_z_level = FALSE
+	//The target z-level of the wipe
+	var/target_z_level = 0
+	//A list of all turfs on the z-leves divided into equal parts depending on how many ticks we have to wipe
+	var/list/wiping_divided_turfs
+	//The turf group we are currently wiping
+	var/wipe_process_num  = 0
+	//Data holder for turf wiping
+	var/datum/data_holder/bluespace_exploration/wipe_data_holder
+	//The queue for z-levels to wipe
+	var/list/z_level_queue
+
+	//=====Other====
+	//Our main bluespace drive
+	var/main_bluespace_drive
+	var/list/bluespace_drives
+
+	//=====Subsystem Fire Tracking=====
+	var/current_part = FIRE_UPDATE_SHIPS
+
+	//eh
+	var/list/ignored_atoms = list(null, /mob/dead, /mob/camera, /mob/dview, /atom/movable/lighting_object)
+
+/datum/controller/subsystem/bluespace_exploration/New()
+	. = ..()
+	//Hello lists (Maximum internal arrays exceeded need for this)
+	//Also this has to be done before SSmapping init
+	bluespace_systems = list()
+	z_level_queue = list()
+	ship_traffic_queue = list()
+	tracked_ships = list()
+	spawnable_ships = list()
+	factions = list()
+	star_systems = list()
+	ruin_templates = list()
+	star_links = list()
+	bluespace_drives = list()
+
+/datum/controller/subsystem/bluespace_exploration/Initialize(start_timeofday)
+	. = ..()
+	//Create factions (These ones are static, although each ship has its own)
+	for(var/faction_datum in subtypesof(/datum/faction))
+		factions[faction_datum] = new faction_datum
+
+/datum/controller/subsystem/bluespace_exploration/fire(resumed = 0)
+	if(current_part == FIRE_UPDATE_SHIPS)
+		if(times_fired % 50 == 0)
+			for(var/ship_key in tracked_ships)
+				var/datum/ship_datum/SD = tracked_ships[ship_key]
+				SD.update_ship()
+				if(QDELETED(SD))
+					tracked_ships -= ship_key
+				if(MC_TICK_CHECK)
+					return
+			//Keep doing this just in case
+			if(CONFIG_GET(flag/bluespace_exploration_random_levels))
+				initiate_queued_warp()
+		current_part = FIRE_START_CLEARING
+	if(MC_TICK_CHECK)
+		return
+	if(!CONFIG_GET(flag/bluespace_exploration_random_levels))
+		return
+	if(current_part == FIRE_START_CLEARING)
+		if(!wiping_z_level && LAZYLEN(z_level_queue))
+			var/first = z_level_queue[1]
+			var/value = z_level_queue[first]
+			wipe_z_level(text2num(first), value)
+			CHECK_TICK
+		current_part = FIRE_CONTINUE_CLEARING
+	if(MC_TICK_CHECK)
+		return
+	if(current_part == FIRE_CONTINUE_CLEARING)
+		if(wiping_z_level)
+			continue_wipe(wipe_data_holder, wiping_divided_turfs, wipe_process_num)
+			wipe_process_num += 1
+	current_part = FIRE_UPDATE_SHIPS
```

Comment:
There is a good reason they are divided into 60 fires rather than going for as much time as the subsystem is allowed to process for:
1: When the server is under low load, the subsystem wont use absolutely all the processing time and make it feel as if under high load.
2: When the server is under high load, the subsystem wont indefinitely load and trap people in hyperspace.